libbeat/beat/event.go

```diff
@@ -228,9 +228,18 @@ func metadataKey(key string) (string, bool) {
 	return "", false
 }
 
-// SetErrorWithOption sets jsonErr value in the event fields according to addErrKey value.
-func (e *Event) SetErrorWithOption(jsonErr mapstr.M, addErrKey bool) {
+// SetErrorWithOption sets the event error field with the message when the addErrKey is set to true.
+// if you want to include the data and field you can pass them as parameters and will be appended into the
+// error as fields with the corresponding name
+func (e *Event) SetErrorWithOption(message string, addErrKey bool, data string, field string) {
 	if addErrKey {
-		e.Fields["error"] = jsonErr
+		errorField := mapstr.M{"message": message, "type": "json"}
+		if data != "" {
+			_, _ = errorField.Put("data", data)
```

Comment:
But upon closer inspection, I think we should not add these two fields into the `error` object because they are not part of [ECS error](https://www.elastic.co/guide/en/ecs/current/ecs-error.html). Perhaps find a way to incorporate them into the `error.message`?

Reply:
I just tried to keep compatibility with the current working code from 

https://github.com/elastic/beats/pull/35920/files#diff-cbf4d8fa4d091014e5fe2a57fac3fa48f8361400b3cb55c2cff307b0a544f6abL125-R125

If we need to change this, we might want to do it as a followup work outside of the work on this PR, WDYT?

Reply:
I agree. I missed that they were already present.